### ISSUE TITLE
Create a Secret synchronization mechamism

### ISSUE THREAD

**Issue by amigus** [thycotic/dsv-k8s#11]
## Background

The injector operates as a Kubernetes Mutating Admissions Webhook. It intercepts submissions of Kubernetes Secrets and "injects" data from DevOps Secrets Vault when the expected metadata>annotations are present. 

## Problem

The injector operates on a Secret upon submission to the server--typically when creating it; however, it cannot keep the secret up-to-date once deployed into the cluster.

## Solution

Create a 'syncer' that will scan the Kubernetes [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) operated on by the injector and update them. 

## Design
 
The syncer will be an executable that accepts the same parameters as the injector minus the host:port plus namespace:

```bash
$ /usr/bin/dsv-syncer -?
flag provided but not defined: -?
Usage of ./dsv-syncer:
  -cert string
        the path of the certificate file in PEM format (default "injector.pem")
  -key string
        the path of the certificate key file in PEM format (default "injector.key")
  -namespace string
        the namespace of the secrets to be synchronized (default is all namespaces)
  -roles string
        the path of JSON formatted roles file (default "roles.json")
```
The user will deploy into the cluster as a Kubernetes [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs) via a Helm [Chart](https://helm.sh/docs/chart_template_guide/getting_started/#charts).
Then the cluster will run it periodically, according to the CronJob schedule.

## Implementation

The 'syncer' will be packaged with the injector as a separate executable.

It should reuse significant portions of the existing injector code. Thus the injector will be refactored; however, it will remain functionally unchanged.

It will be a Golang executable (like the injector), so it can be run "manually," as part of an operator, etc., i.e., it is helpful with or without the CronJob deployment.

The chart that deploys the syncer will be separate from the injector chart.
The `values.yaml` will allow the user to set parameters to control the CronJob schedule, namespace, etc.

### Limitations

- It will _not_ allow for each secret to having its update interval; rather, the update interval will be cluster-wide. However, a future enhancement could support an additional annotation representing a custom interval that is a multiple of the cluster-wide update interval. For example, the cluster updates every 30 seconds but the secret only updates every five minutes. That will reduce queries to DevOps Secrets Vault depending on how often users use it.
- It will require a Role and associated RoleBinding, allowing the ServiceAccount running syncer to read and update _all_ secrets.
- It polls, so it cannot "instantly" update a secret on change on the DevOps Secrets Vault side.
- It is a one-way sync; it will not store any changes from the Kubernetes side in DevOps Secrets Vault.

## TL;DR
Create a 'syncer' that runs as a Kubernetes CronJob to update injector-mutated Kubernetes Secrets periodically.
